server/server.go

```diff
@@ -469,43 +469,81 @@ func New(
 	return s, nil
 }
 
-func (s *Server) getHostIP(configIP string) net.IP {
+func (s *Server) getHostIPs(configIP string) []string {
 	// emulate kubelet behavior of choosing hostIP
 	// ref: k8s/pkg/kubelet/nodestatus/setters.go
+	rootIP := ""
 
 	// use configured value if set
 	if hostIP := net.ParseIP(configIP); hostIP != nil {
-		return hostIP
+		rootIP = hostIP.String()
 	}
 
 	// Otherwise use kubernetes utility to choose hostIP
 	// there exists the chance for both hostIP and err to be nil, so check both
-	if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
-		return hostIP
+	if rootIP == "" {
+		if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
+			rootIP = hostIP.String()
+		}
 	}
 
 	// attempt to find an IP from the hostname
-	if hostname, err := os.Hostname(); err == nil {
-		if hostIP := net.ParseIP(hostname); hostIP != nil && validateHostIP(hostIP) == nil {
-			return hostIP
+	if rootIP == "" {
+		if hostname, err := os.Hostname(); err == nil {
+			if hostIP := net.ParseIP(hostname); hostIP != nil && validateHostIP(hostIP) == nil {
```

Comment:
validateHostIP is v4 only. What if the hostname is a v6 IP?

Reply:
I removed it that it also allow IPv6 addresses.